parser/parser_system1.c

```diff
@@ -376,6 +377,23 @@ static int parse_bsp(struct pv_state *s, char *value, int n)
 	if (pv_config_get_system_init_mode() == IM_APPENGINE)
 		return 1;
 
+	count = pv_json_get_key_count(value, "bsp/run.json", tv, tc);
```

Comment:
what is the purpose of the changes in this function?

Reply:
i will try to make this better...

Reply:
i believe new approached pushed is in your spirit.... check it